lib/Manager.php

```diff
@@ -908,6 +910,12 @@ public function getChangelogRoom(string $userId): Room {
 	 * @return Room
 	 */
 	public function createRoom(int $type, string $name = '', string $objectType = '', string $objectId = ''): Room {
+
+		// Force group room if public room was requested and public rooms are disallowed.
+		if (($type === Room::TYPE_PUBLIC) && ($this->config->getAppValue('spreed', 'public_rooms_allowed', 'yes') === 'no')) {
+			$type = Room::TYPE_GROUP;
```

Comment:
Not sure if we should simply convert it. Or just error with a 400 Bad Request and a message the clients can display.

Reply:
Conversion was choosen because spreed code ignores errors returned by modified functions and this seems to be safer. Conversion should never happen because function should be not available in UI also (and if bad guys will try to modify POST, will get group conversation which should not hurt regular users).

Reply:
That assumes all clients got updated in the meantime :)

Reply:
Sorry, we use web UI only - feel free to extend it for other requires areas if necessarry (not experienced enough in nc devel to propose it now).

Reply:
You can fix it with one step by moving it to `RoomService::createConversation` and throwing an exception or basically adjust this list here:
https://github.com/nextcloud/spreed/blob/220089657318501ff9b08941349b27a79b0f7a1b/lib/Service/RoomService.php#L141-L147

Reply:
`type` is property of `Room` not `RoomService`  so verification should take place in `Room` class to avoid future problems (i.e. code outside of `RoomService` will create/change room of disabled type). Seems that now `Room::setType` and `Room` constructor are the best places for this verification but code using it should properly handle errors when disabled type is passed. Please let me know if its possible to just throw exception in `Room` methods mentioned above without messing with room type conversions (there should not be any other trash left after such exception - not sure if it would be safe now).